main/tray.js

```diff
@@ -85,6 +91,15 @@ function setupIpcEventListeners (contextMenu) {
     getItemById('checkingForUpdates').visible = false
   })
 
+  ipcMain.on(ipcMainEvents.ACTIVITY_LOGGED, () => {
+    assert(tray)
+
+    const state = saturn.isOnline() ? 'on' : 'off'
+    const image = nativeImage.createFromPath(icon(state))
+    image.setTemplateImage(true)
+    tray.setImage(image)
```

Comment:
How expensive is it to create a new native image? Can we create these two images (on & off) at start time, keep them around for the entire life of the app, and then just select the right image based on the state in this code?

Something along the following lines:

```js
    const trayIcon = saturn.isOnline() ? iconForStateOn : iconForStateOff;
    tray.setImage(trayIcon)
```

Reply:
great suggestion!